### DISCUSSION TITLE
List Filter

### DISCUSSION THREAD

**Discussion by Sarveshgithub** [Sarveshgithub/sfdc-lwc-lightning-datatable#165] [Show and tell]
This discussion is for List Filter - which is a new feature, this is a big feature. 
- Validate functionality
- Add feature
- Complete feature

----------------------

Hey @mickaelgudin I have implemented the list filter functionality, the code is in the'listFilter' branch, and only a few features are pending, I have added them below, can you please contribute to this? feel free to add comments, if all is clear then create Issue  ( enhancement )

**Comment by Sarveshgithub** [comment]
**Clone Filter** - can be the same code from - New Filter Button

**Reply by Sarveshgithub** [reply]
@mickaelgudin

**Comment by Sarveshgithub** [comment]
**Filter Fields** - As of now all fields are showing, we need to filter out fields based on input fields

**Reply by Sarveshgithub** [reply]
@mickaelgudin

**Comment by mickaelgudin** [comment]
I believe the filters should be optional, since we create an object..., we could split the repository in two parts :

- one without filters (branch master)
- one with filters (branch listFilter)

The remaning features can wait, you can redirect the users that want this feature(in the readme) to the branch listFilter (make sure to merge master into the listFilter).

**I'm focusing my efforts on the picklist feature (before I promote the repository on Linkedin and Medium)**

@Sarveshgithub  What do you think about splitting the project and focusing on picklist before promoting the repo ?

**Reply by Sarveshgithub** [reply]
hmm, fine no problem. we'll merge the list filter feature afterward picklist
